src/addons/retry.rs

```diff
@@ -0,0 +1,68 @@
+use bevy::prelude::*;
+
+use crate::{LocalCommand, Process, ProcessCompleted};
+
+#[derive(Debug, Component)]
+pub enum Retry {
+    Attempts(usize),
+}
+
+/// Retry failed processes according to the Retry component.
+///
+/// Processes without the Retry component are ignored.
+/// The Retry component is removed from the entity when retries are done.
+pub(crate) fn retry_failed_process(
+    mut commands: Commands,
+    query: Query<(&LocalCommand, &Retry)>,
+    mut process_completed_events: EventReader<ProcessCompleted>,
+) {
+    for process_completed_event in process_completed_events.read() {
+        if process_completed_event.exit_status.success() {
+            continue;
+        }
+        if let Ok((current_local_command, retry)) = query.get(process_completed_event.entity) {
+            match retry {
+                Retry::Attempts(retries) => {
+                    if let Some(mut entity_commands) =
+                        commands.get_entity(process_completed_event.entity)
+                    {
+                        // Remove Retry component and skip if no more retries left
+                        if *retries < 1 {
+                            entity_commands.remove::<Retry>();
+                            continue;
+                        }
+
+                        // Remove both the Process and LocalCommand
+                        entity_commands.remove::<(Process, LocalCommand)>();
```

Comment:
I think I'd prefer if we'd only remove this if the retries are over.
The new components would only be spawned in the next exclusive system which applies the commands, not sure if that could cause issues here.

We should be able to modify them in place by dereferencing the pointer I think

Reply:
Makes sense. Seems to work.